vpc.go

```diff
@@ -0,0 +1,269 @@
+		RoutesPerRouteTableQuota:         prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_routesperroutetable_quota"), "The quota of routes per routetable", []string{"aws_region"}, nil),
+		RoutesPerRouteTableUsage:         prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_routesperroutetable_usage"), "The usage of routes per routetable", []string{"aws_region", "vpcid", "routetableid"}, nil),
+		InterfaceVpcEndpointsPerVpcQuota: prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_interfacevpcendpointspervpc_quota"), "The quota of interface vpc endpoints per vpc", []string{"aws_region"}, nil),
+		InterfaceVpcEndpointsPerVpcUsage: prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_interfacevpcendpointspervpc_usage"), "The usage of interface vpc endpoints per vpc", []string{"aws_region", "vpcid"}, nil),
+		RouteTablesPerVpcQuota:           prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_routetablespervpc_quota"), "The quota of route tables per vpc", []string{"aws_region"}, nil),
+		RouteTablesPerVpcUsage:           prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_routetablespervpc_usage"), "The usage of route tables per vpc", []string{"aws_region", "vpcid"}, nil),
+		IPv4BlocksPerVpcQuota:            prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_ipv4blockspervpc_quota"), "The quota of ipv4 blocks per vpc", []string{"aws_region"}, nil),
+		IPv4BlocksPerVpcUsage:            prometheus.NewDesc(prometheus.BuildFQName(namespace, "", "vpc_ipv4blockspervpc_usage"), "The usage of ipv4 blocks per vpc", []string{"aws_region", "vpcid"}, nil),
+		logger:                           logger,
+	}
+}
+
+func (e *VPCExporter) Collect(ch chan<- prometheus.Metric) {
+	ec2Svc := ec2.New(e.sess)
+	serviceQuotaSvc := servicequotas.New(e.sess)
+
+	e.collectVpcsPerRegionQuota(ec2Svc, serviceQuotaSvc, ch)
+	e.collectVpcsPerRegionUsage(ec2Svc, ch)
+	// TODO: Getting all VPCs is probably too much?
+	e.collectRoutesTablesPerVpcQuota(ec2Svc, serviceQuotaSvc, ch)
+	e.collectInterfaceVpcEndpointsPerVpcQuota(ec2Svc, serviceQuotaSvc, ch)
+	e.collectSubnetsPerVpcQuota(ec2Svc, serviceQuotaSvc, ch)
+	e.collectIPv4BlocksPerVpcQuota(ec2Svc, serviceQuotaSvc, ch)
+	allVpcs, err := ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{})
+	if err != nil {
+		level.Error(e.logger).Log("msg", "Call to DescribeVpcs failed", "region", *e.sess.Config.Region, "err", err)
+	} else {
+		for i, _ := range allVpcs.Vpcs {
+			e.collectSubnetsPerVpcUsage(ec2Svc, ch, allVpcs.Vpcs[i])
+			e.collectInterfaceVpcEndpointsPerVpcUsage(ec2Svc, ch, allVpcs.Vpcs[i])
+			e.collectRoutesTablesPerVpcUsage(ec2Svc, ch, allVpcs.Vpcs[i])
+			e.collectIPv4BlocksPerVpcUsage(ec2Svc, ch, allVpcs.Vpcs[i])
+		}
+	}
+	// TODO: Do we pass the RouteTable or do we just get *all* again?
```

Comment:
I don't think we have all that many RouteTables per account, so I believe this should be fine.